lib/rules/no-promise-executor-return.js

```diff
@@ -59,6 +60,72 @@ function isPromiseExecutor(node, scope) {
         isGlobalReference(parent.callee, getOuterScope(scope));
 }
 
+/**
+ * Checks if the given node is a void expression.
+ * @param {ASTNode} node The node to check.
+ * @returns {boolean} - `true` if the node is a void expression
+ */
+function expressionIsVoid(node) {
+    return node.type === "UnaryExpression" && node.operator === "void";
+}
+
+/**
+ * Fixes the linting error by prepending "void " to the given node
+ * @param {Object} sourceCode context given by context.sourceCode
+ * @param {ASTNode} node The node to fix.
+ * @param {Object} fixer The fixer object provided by ESLint.
+ * @returns {Array<Object>|Object} - An array of fix objects or fix to apply to the node.
+ */
+function voidPrependFixer(sourceCode, node, fixer) {
```

Comment:
This works well, except in cases where the token before it inserts `void` was right after `return`:

```js
/*eslint no-promise-executor-return: ["error", { allowVoid: true }]*/

new Promise(r => { return(a || b) ? r(1) : r(2) });

new Promise(r => { return(a ? r(1) : r(2)) });

```

After the prepend void suggestions are applied:

```js
/*eslint no-promise-executor-return: ["error", { allowVoid: true }]*/

new Promise(r => { returnvoid ((a || b) ? r(1) : r(2)) });

new Promise(r => { returnvoid (a ? r(1) : r(2)) });

```

We can check if the tokens were adjacent (return token's range[1] === that token's range[0]) and in that case add one space before `void`.